terraform/catalogue_api/main.tf

```diff
@@ -0,0 +1,45 @@
+module "catalogue_api_prod" {
+  source = "../stack"
+
+  environment_name = "prod"
+  container_images = {
+    search = "${local.search_repository}:env.prod"
+    items  = "${local.items_repository}:env.prod"
+  }
+  desired_task_counts = {
+    search = 3
+    items  = 3
+  }
+
+  vpc_id                   = local.vpc_id
+  private_subnets          = local.private_subnets
+  elastic_cloud_vpce_sg_id = local.elastic_cloud_vpce_sg_id
+  cluster_arn              = aws_ecs_cluster.catalogue_api.arn
+
+  providers = {
+    aws.dns = aws.dns
+  }
+}
```

Comment:
Do we want to have prod/stage in the same configuration? Historically we had them separate, so you could test changes in staging without blowing up prod.

Reply:
I think - as per some discussion with @kenoir in Slack - we do want them to be shared like this as it is more consistent with our other services, and it's easier to keep them in sync. You can still test them separately (as I have been doing) with a `terraform apply -target=module.catalogue_api_stage`

Reply:
\*nod* I must have missed that discussion.

Did you consider workspaces? https://www.terraform.io/docs/language/state/workspaces.html I haven't used them, but Digirati used them for the identity API and they sound interesting.

Reply:
I don't think we want to use workspaces.

From the docs:

```
In particular, organizations commonly want to create a strong separation between multiple deployments of the same infrastructure serving different development stages (e.g. staging vs. production) or different internal teams. In this case, the backend used for each deployment often belongs to that deployment, with different credentials and access controls. Named workspaces are not a suitable isolation mechanism for this scenario.
```

Reply:
I did not - it seems like they'd be very much equivalent to what they're doing albeit a bit more painful to configure differences (https://www.terraform.io/docs/language/state/workspaces.html#current-workspace-interpolation). I'll probably leave them out for now because I'm not sure how shifting some of the old state across would work with them (aka, better the devil you know)

Reply:
@kenoir I was going to quote that but then I realised that holding the stacks in the same configuration suffers the exact same drawback 🤷‍♂️